src/discord/commands/book.ts

```diff
@@ -0,0 +1,229 @@
+    //Function finds all available timeblocks for a given date
+
+    const currentDate = new Date();
+    let nextHour = '00';
+
+    if (currentDate.toLocaleDateString().split('/').reverse().join('-') === selectedDate) {
+        nextHour = String(currentDate.getHours() + 1);
+
+        if (nextHour.length === 1) {
+            nextHour = `0${nextHour}`;
+        }
+    }
+
+    const startDate = DateTime.fromISO(String(`${selectedDate}T${nextHour}:00`), { zone: 'America/Toronto' });
+    const endDate = DateTime.fromISO(String(`${selectedDate}T23:00`), { zone: 'America/Toronto' });
+
+    const bookedTimeblocks = await Timeblock.find({
+        sectionId: sectionInformation._id,
+        startsAt: { $gte: startDate.toJSDate() },
+        endsAt: { $lte: endDate.toJSDate() },
+    });
+
+    const timeBlocks = [];
+    let currHourStart = startDate;
+    let currHourEnd = currHourStart.plus({ hours: 1 });
+
+    // Iterate through the hours in the given time range
+    while (currHourStart < endDate) {
+        // Finds schedule start and end for the day that the current hour falls on
+        const currWeekDay = currHourStart.weekday;
+        let scheduleDayStart = 0;
+        let scheduleDayEnd = 0;
+        for (const day of roomInformation.schedule) {
+            if (day.dayOfWeek + 1 === currWeekDay) {
+                scheduleDayStart = day.start;
+                scheduleDayEnd = day.end;
+            }
+        }
+
+        // Checks if the hour falls under an open time for the room and if it does adds hour to the list of available times
+        if (!roomInformation.closed && currHourStart.weekday === currHourEnd.weekday && currHourStart.hour >= scheduleDayStart && currHourEnd.hour <= scheduleDayEnd) {
+            // Check for time block in database
+            const bookedTimeBlockFound = bookedTimeblocks.find((bookedTimeBlock) => bookedTimeBlock.startsAt.getTime() === currHourStart.toMillis());
+
+            if (bookedTimeBlockFound) {
+                const newTimeBlock = {
+                    startsAt: currHourStart.toJSDate(),
+                    endsAt: currHourEnd.toJSDate(),
+                    availableCapacity: sectionInformation.capacity - bookedTimeBlockFound.users.length,
+                };
+                timeBlocks.push(newTimeBlock);
+            } else {
+                const newTimeBlock = {
+                    startsAt: currHourStart.toJSDate(),
+                    endsAt: currHourEnd.toJSDate(),
+                    availableCapacity: sectionInformation.capacity,
+                };
+                timeBlocks.push(newTimeBlock);
+            }
+        }
+
+        // Increments hour start and end for next iteration
+        currHourStart = currHourStart.plus({ hours: 1 });
+        currHourEnd = currHourEnd.plus({ hours: 1 });
+    }
+    return timeBlocks;
+}
+
+function timeConversion(timeObject: Date) {
```

Comment:
Chances are you can replace this function with Luxon too

Reply:
It might be better to rename this method to `formatTime` or something for better clarity